src/prefect/tasks/monte_carlo/client.py

```diff
@@ -0,0 +1,256 @@
+import json
+import pendulum
+import requests
+from typing import Any, Dict, List
+
+from prefect.utilities.logging import get_logger
+
+
+class MonteCarloClient:
+    def __init__(
+        self,
+        api_key_id: str,
+        api_token: str,
+    ) -> None:
+        self.api_key_id = api_key_id
+        self.api_token = api_token
+        self.logger = get_logger()
+        self._api_url = "https://api.getmontecarlo.com/graphql"
+
+    def graphql(self, query: str, variables: dict = None) -> Dict[str, Any]:
+        response = requests.post(
+            url=self._api_url,
+            json=dict(query=query, variables=variables),
+            headers={
+                "x-mcd-id": self.api_key_id,
+                "x-mcd-token": self.api_token,
+                "Content-Type": "application/json",
+            },
+        )
+        response.raise_for_status()
+        response = json.loads(response.text)
+        self.logger.info("Response: %s", response)
+        return response
+
+    def get_warehouse_id_by_name(self, warehouse_name: str) -> str:
+        response = self.graphql(
+            query="""
+                    {
+                        getUser {
+                            account {
+                                warehouses {
+                                    name
+                                    uuid
+                                }
+                            }
+                        }
+                    }
+                    """
+        )
+        warehouses = response["data"]["getUser"]["account"]["warehouses"]
+        id_ = [w for w in warehouses if w["name"] == warehouse_name][0].get("uuid")
+        return id_
+
+    def get_resources(self) -> List[Dict[str, Any]]:
+        response = self.graphql(
+            query="""
+                    query {
+                        getResources {
+                            name
+                            type
+                            id
+                            uuid
+                            isDefault
+                            isUserProvided
+                        }
+                    }
+                    """
+        )
+        return response["data"]["getResources"]
+
+    def get_table_mcon(self, full_table_id: str, warehouse_name: str) -> str:
+        response = self.graphql(
+            query="""
+                    query($dwh_id: UUID!, $full_table_id: String!) {
+                        getTable(dwId: $dwh_id,
+                            fullTableId: $full_table_id) {
+                            mcon
+                        }
+                    }
+                    """,
+            variables=dict(
+                dwh_id=self.get_warehouse_id_by_name(warehouse_name),
+                full_table_id=full_table_id,
+            ),
+        )
+        return response["data"]["getTable"]["mcon"]
```

Comment:
I recommend raising a custom exception (e.g. `TableNotFound`) when the specified table cannot be found.

Reply:
also here: I will entirely remove this method because it's no longer necessary, this will make things cleaner 👍